Task/add missing task docs (#484)
tandemdude/hikari-lightbulb: fragments/484.documentation.md

> Added missing Documentation for the `Task` class and secondary `task` decorators, stating that setting `max_failures` or `max_invocations` to a negative number will disable that behaviour.

Reviewer: This could be made shorter -
```suggestion
Document the behaviour of `max_failures` and `max_invocations` in tasks and how to disable each.
```
